### ISSUE TITLE
Lint from stdin

### ISSUE THREAD

**Issue by hypothesist** [coala-analyzer/coala#2540]

**Comment by sils1297** [comment]
where?who?how?

**Comment by hypothesist** [comment]
Lint from another program's pipe output for instance

**Comment by sils1297** [comment]
Right, how do we do this? How does the user do this? Apply the section to stdin? The default section?

**Comment by hypothesist** [comment]
@sils1297 probably the coafile that is found? IMO we just need to allow something like `-f -` because `-` generally means stdin.

**Comment by hypothesist** [comment]
To anyone who's picking this up: this is basically to allow stuff like `cat setup.py | coala`

**Comment by Makman2** [comment]
wait do we actually want that? imo we don't need to integrate that into `Linter`, as it's possible to write a custom (python-)script/executable and provide that as the exe-name. The script would then take care of invoking the first tool and pass its output to the other tool.

I can't imagine a case where this is useful currently^^

Example for a python script taking care of this:
```
# the script:
from coalib.misc.Shell import run_shell_command  # <-- or use subprocess module directly
stdout1, stderr1 = run_shell_command(["tool1", "arg1"])
stdout2, stderr2 = run_shell_command(["tool2", "--stdin"], stdin=stdout1)
print(stdout2 if stdout2 else stderr2)

# the bear:
import sys

@linter(executable=sys.executable, ...)
class MyBear:
    @staticmethod
    def create_arguments(filename, file, config_file):
        return "scriptname",
```

**Comment by adtac** [comment]
@Makman2 We could build this into coala? Like `coala --stdin`? Since we recently merged coala-json, coala-format, coala-ci into coala as flag options, something like `coala-stdin` could be a flag option too.

**Comment by sils** [comment]
certainly, that would probably lead to the main function creating a
temporary file and setting the `--files` argument implicitly to that, hm? A
lot of linter bears rely on the file being physically existent. Also this
would be super simple to implement.

**Comment by adtac** [comment]
Yeah, a temp file makes sense (we have `make_temp` too). Change this to difficulty/low then?

**Comment by Makman2** [comment]
seems I've misunderstood, nvm :3
